modules/fbs-core/api/src/main/scala/de/thm/ii/fbs/services/checker/excel/ExcelCheckerService.scala

```diff
@@ -126,6 +150,43 @@ class ExcelCheckerService extends CheckerService {
     }
   }
 
+  private def buildTaskResultText(result: CheckResultTask, task: ExcelMediaInformation) = {
+    if (task.checkFields.isEmpty) {
+      buildLegacyTaskResultText(result, task)
+    } else {
+      result.checkResult.zip(task.checkFields)
+        .filter(shouldBuildCheckResult)
+        .map(c => buildCheckResult(c._1, c._2, task))
+        .mkString("\n")
+    }
+  }
+
+  private def buildLegacyTaskResultText(result: CheckResultTask, task: ExcelMediaInformation) = {
+    if (result.checkResult.head.errorMsg.nonEmpty) {
+      f"${task.name}: ${result.checkResult.head.errorMsg}"
+    } else {
+      f"${task.name}: Die Zelle/-n '${result.checkResult.head.invalidFields.mkString(", ")}' enthalten nicht das korrekte Ergebnis"
+    }
+  }
+
+  private def buildCheckResult(result: CheckResult, check: ExcelMediaInformationCheck, task: ExcelMediaInformation) = {
+    if (result.errorMsg.nonEmpty) {
+      f"${task.name}: ${result.errorMsg}"
+    } else {
+      val errorMsg = if (check.errorMsg.nonEmpty) f"${check.errorMsg}" else ""
+
+      if (check.hideInvalidFields) {
+        f"${task.name}: $errorMsg"
+      } else {
+        f"${task.name}: Die Zelle/-n '${result.invalidFields.mkString(", ")}' enthalten nicht das korrekte Ergebnis. $errorMsg"
```

Comment:
"Die Zellen ... enthalten nicht das korrekte Ergebnis."
There is no need for the alternative plural of "Zelle" if "enthalten" is not in the right form either.